Add a generic GeocoderCombo component

chrismayer opened the pull request:
This adds a generic `GeocoderCombo` component similar to those we had in previous GeoExt versions.

To sum it up it is a combo box that handles results from a geocoding service. By default it uses OSM Nominatim, but the component offers all config options to overwrite in order to use other custom services.

The component is only usable with the classic toolkit. An appropriate `onClassCreated` callback for not breaking modern apps is implemented. Also a warning is placed in the API docs.

weskamm commented on src/form/field/GeocoderComboBox.js:
```diff
@@ -0,0 +1,327 @@
+
+        if (!me.locationLayer) {
+            me.locationLayer = new ol.layer.Vector({
+                source: new ol.source.Vector()
```
~~Maybe we can add a user-configurable style to the layer?~~
But its not really necessary as the user can configure the whole layer by himself, so just ignore the comment

chrismayer replied:
Thanks for the hint. I added a config property so you can influence the style even if the layer is auto-created. Might be handy at some point.

weskamm commented on test/spec/GeoExt/form/field/GeocoderComboBox.test.js:
```diff
@@ -0,0 +1,148 @@
+            it('centers the map to coordinate', function() {
+                var record = {
+                    get: function() {
+                        return [0, 0];
```
maybe you can change the coordinates to something else, as [0,0] are the defaults of the map already

chrismayer replied:
Done, see 0cd0dc59164c4d5a0

chrismayer commented:
Thanks for the review @weskamm. I addressed your points. Would you be available for another quick review of the changes?

The failing CI is unrelated to these changes IMHO (Something Java exception in Sencha CMD, executed in  `./ci/create-sencha-package.sh`.

weskamm commented:
All looking good, feel free to merge!

chrismayer commented:
Thanks again for the review @weskamm! Will merge now.

chrismayer commented:
Right @marcjansen combining with `||` would be correct, but this isn't the problem. I fixed it locally. The root of the problem is as described in #504. I had a quick chat with @weskamm. He has detailed information and will have a look at this.

weskamm commented:
Issue fixed by https://github.com/geoext/geoext3/pull/507
